src/components/App.js

```diff
@@ -0,0 +1,31 @@
+import React from 'react'; // eslint-disable-line no-unused-vars
+import Display from './Display'; // eslint-disable-line no-unused-vars
+import ButtonPanel from './ButtonPanel'; // eslint-disable-line no-unused-vars
+import calculate from '../logic/calculate'; // eslint-disable-line no-unused-vars
+import './App.css';
+
+class App extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      total: null,
+      next: null,
+      operation: null,
+    };
+  }
+
+  clickHandler = (button) => {
+    this.setState(calculate(this.state, button));
+  };
+
+  render() {
+    return (
+      <div className="App">
+        <Display result={this.state.next || this.state.total || '0'} />
```

Comment:
hey, you don't really need the `0`. you already have defaultProps of display set to `0`. just next or total is fine.